File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/AuthorColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SexEnum } from "../Modules/Common.Enums.SexEnum";
import { AuthorRow } from "./AuthorRow";

export interface AuthorColumns {
    AuthorName: Column<AuthorRow>;
    Sex: Column<AuthorRow>;
    Email: Column<AuthorRow>;
    UpdateTime: Column<AuthorRow>;
}

export class AuthorColumns extends ColumnsBase<AuthorRow> {
    static readonly columnsKey = 'BookManage.Author';
    static readonly Fields = fieldsProxy<AuthorColumns>();
}

[SexEnum]; // referenced types

File: LMIS/LMIS.Web/Modules/BookManage/Author/AuthorColumns.cs
using LMIS.Modules.Common.Enums;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Author")]
[BasedOnRow(typeof(AuthorRow), CheckNames = true)]
public class AuthorColumns
{
    [EditLink, AlignRight]
    public long AuthorId { get; set; }
    [EditLink]
    public string AuthorName { get; set; }
    public SexEnum Sex { get; set; }
    [Width(250)]
    public string Email { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}